Sources/PointFree/RunnerTasks/WelcomeEmails.swift

```diff
@@ -12,49 +12,60 @@ import Prelude
 import Styleguide
 import Views
 
-public func sendWelcomeEmails() -> EitherIO<Error, Prelude.Unit> {
+public func sendWelcomeEmails() async throws {
+  @Dependency(\.continuousClock) var clock
   @Dependency(\.database) var database
 
-  let emails = EitherIO {
+  await notifyError(subject: "Welcome emails failed") {
     async let emails1 = database.fetchUsersToWelcome(1).map(welcomeEmail1)
     async let emails2 = database.fetchUsersToWelcome(2).map(welcomeEmail2)
-    async let emails3 = database.fetchUsersToWelcome(3).map(welcomeEmail3)
-    return try await emails1 + emails2 + emails3
-  }
-  .debug { "📧: Sending \($0.count) welcome emails..." }
-
-  let delayedSend = { email in
-    EitherIO { try await send(email: email) }
-      .delay(.milliseconds(200))
-      .retry(maxRetries: 3, backoff: { .seconds(10 * $0) })
-  }
+    async let users = database.fetchUsersToWelcome(3)
+    let emails3 = try await users.map(welcomeEmail3)
+    if try await !users.isEmpty {
+      _ = try await database.incrementEpisodeCredits(users.map(\.id))
+    }
 
-  return
-    emails
-    .flatMap(map { email in delayedSend(email).map(const(email)) } >>> sequence)
-    .flatMap { (emails: [Email]) -> EitherIO<Error, SendEmailResponse> in
-      let stats =
-        emails
-        .reduce(into: [String: [EmailAddress]]()) { dict, email in
-          dict[email.subject, default: []].append(contentsOf: email.to)
+    let emails = try await emails1 + emails2 + emails3
+    print("📧: Sending \(emails.count) welcome emails...")
+    var sentEmails: [SendEmailResponse] = []
+    for email in emails {
+      do {
+        try await sentEmails.append(send(email: email))
+        try await clock.sleep(for: .milliseconds(200))
+      } catch {
+        do {
+          try await clock.sleep(for: .seconds(10))
+          try await sentEmails.append(send(email: email))
+        } catch {
+          do {
+            try await clock.sleep(for: .seconds(10))
+            try await sentEmails.append(send(email: email))
+          } catch {
+            try await clock.sleep(for: .seconds(10))
+            try await sentEmails.append(send(email: email))
+          }
```

Comment:
This just recaptures the `retry` operator...we could prob write an `async`-specific one, but this was easy for now.